ivy/utils/_importlib.py

```diff
@@ -1,11 +1,95 @@
+import os
 import sys
+from typing import List, Set
+from pathlib import Path
 from importlib.util import resolve_name, module_from_spec
 from ivy.utils.backend import ast_helpers
 
 
 import_cache = {}
 path_hooks = []
 
+ivy_path_abs = Path(sys.modules["ivy"].__file__).parents[1]
+
+# Note that any modules listed as 'to skip' should not depend on the Ivy backend state.
+# If they do, the behavior of ivy.with_backend is undefined and may not function as
+# Expected.
+MODULES_TO_SKIP = ["ivy.compiler"]
+MODULES_TO_EXCLUDE = []
+
+
+def _get_modules(absolute_name: str) -> Set[str]:
+    """Get all Ivy modules and sub-packages in a module
+
+    Parameters
+    ----------
+    absolute_name
+        Full absolute name of the module
+
+    Returns
+    -------
+        A set of modules and sub-packages found
+    """
+    if not absolute_name.startswith("ivy."):
+        raise ValueError("Module name must be specified using full namespace.")
+
+    name_to_path = absolute_name.replace(".", os.path.sep)
+    module_path = os.path.join(ivy_path_abs, name_to_path)
+
+    modules = set()
+    modules.add(absolute_name)
+
+    for root, dirs, files in os.walk(module_path):
+        if root.endswith("__"):
+            continue
+        common = os.path.commonpath([root, module_path])
+        full_name = absolute_name + root[len(common) :].replace(os.path.sep, ".")
+        for mod_name in files:
+            if mod_name.startswith("__") or not mod_name.endswith(".py"):
+                continue
+            modules.add(f"{full_name}.{mod_name[:-3]}")
+        for dir_name in dirs:
+            if dir_name.startswith("__"):
+                continue
+            modules.add(f"{full_name}.{dir_name}")
```

Comment:
same here, could be 
```python
modules.update([f"{full_name}.{dir_name}" for dir_name in dirs if not dir_name.startswith("__")])
```
Again, debatable if this is more readable (I personally think it is)